tasks/cops_documentation.rake

```diff
@@ -0,0 +1,298 @@
+  def print_cops_of_department(cops, department, config)
+    selected_cops = cops_of_department(cops, department).select do |cop|
+      cop.to_s.start_with?('RuboCop::Cop::RSpec')
+    end
+    return if selected_cops.empty?
+    content = "# #{department}\n".dup
+    selected_cops.each do |cop|
+      content << print_cop_with_doc(cop, config)
+    end
+    file_name = "#{Dir.pwd}/manual/cops_#{department.downcase}.md"
+    File.open(file_name, 'w') do |file|
+      puts "* generated #{file_name}"
+      file.write(content.strip + "\n")
+    end
+  end
+  # rubocop:enable Metrics/AbcSize
+  # rubocop:enable Metrics/MethodLength
+
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
+  def print_cop_with_doc(cop, config)
+    t = config.for_cop(cop)
+    non_display_keys = %w[Description Enabled StyleGuide Reference]
+    pars = t.reject { |k| non_display_keys.include? k }
+    description = 'No documentation'
+    examples_object = []
+    YARD::Registry.all(:class).detect do |code_object|
+      next unless RuboCop::Cop::Badge.for(code_object.to_s) == cop.badge
+
+      description = code_object.docstring unless code_object.docstring.blank?
+      examples_object = code_object.tags('example')
+    end
+    cops_body(config, cop, description, examples_object, pars)
+  end
+  # rubocop:enable Metrics/AbcSize
+  # rubocop:enable Metrics/MethodLength
+
+  # rubocop:disable Metrics/AbcSize
+  # rubocop:disable Metrics/MethodLength
+  def table_of_content_for_department(cops, department)
+    selected_cops = cops_of_department(cops, department.to_sym).select do |cop|
+      cop.to_s.start_with?('RuboCop::Cop::RSpec')
+    end
+    return if selected_cops.empty?
+    type_title = department[0].upcase + department[1..-1]
+    filename = "cops_#{department.downcase}.md"
+    content = "#### Department [#{type_title}](#{filename})\n\n".dup
+    selected_cops.each do |cop|
+      anchor = cop.cop_name.sub('/', '').downcase
+      content << "* [#{cop.cop_name}](#{filename}##{anchor})\n"
+    end
+
+    content
+  end
+  # rubocop:enable Metrics/AbcSize
+  # rubocop:enable Metrics/MethodLength
+
+  def print_table_of_contents(cops)
+    path = "#{Dir.pwd}/manual/cops.md"
+    original = File.read(path)
+    content = "<!-- START_COP_LIST -->\n".dup
+
+    content << table_contents(cops)
+
+    content << "\n<!-- END_COP_LIST -->"
+
+    content = original.sub(
+      /<!-- START_COP_LIST -->.+<!-- END_COP_LIST -->/m, content
+    )
+    File.write(path, content)
+  end
+
+  def table_contents(cops)
+    cops
+      .departments
+      .map(&:to_s)
+      .sort
+      .map { |department| table_of_content_for_department(cops, department) }
+      .reject(&:nil?)
+      .join("\n")
+  end
+
+  def assert_manual_synchronized
+    # Do not print diff and yield whether exit code was zero
+    sh('git diff --quiet manual') do |outcome, _|
+      return if outcome
+
+      # Output diff before raising error
+      sh('git diff manual')
```

Comment:
The default pager for `git diff` is `less`. And `less` doesn’t quit, which then causes the CI to hang.

E.g. `sh('PAGER=cat git diff manual')` should fix it.

Reply:
I think we don't need this assert, since we have such check in `confirm_documentation` as well

Reply:
True, that is exactly what `rake confirm_documentation` does.